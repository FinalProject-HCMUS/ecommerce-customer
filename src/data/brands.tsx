export const brands = [
  { name: 'Versace', logo: 'VERSACE' },
  { name: 'Zara', logo: 'ZARA' },
  { name: 'Gucci', logo: 'GUCCI' },
  { name: 'Prada', logo: 'PRADA' },
  { name: 'Calvin Klein', logo: 'Calvin Klein' },
];
