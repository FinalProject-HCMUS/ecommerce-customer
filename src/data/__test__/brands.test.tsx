import { brands } from '../brands';

describe('brands data', () => {
  test('should export an array of brands', () => {
    expect(Array.isArray(brands)).toBe(true);
    expect(brands.length).toBeGreaterThan(0);
  });

  test('each brand should have name and logo properties', () => {
    brands.forEach((brand) => {
      expect(brand).toHaveProperty('name');
      expect(typeof brand.name).toBe('string');
      expect(brand).toHaveProperty('logo');
      expect(typeof brand.logo).toBe('string');
    });
  });

  test('should contain specific brands', () => {
    const brandNames = brands.map((b) => b.name);
    expect(brandNames).toEqual(
      expect.arrayContaining([
        'Versace',
        'Zara',
        'Gucci',
        'Prada',
        'Calvin Klein',
      ])
    );
  });

  test('should match snapshot', () => {
    expect(brands).toMatchSnapshot();
  });
});
