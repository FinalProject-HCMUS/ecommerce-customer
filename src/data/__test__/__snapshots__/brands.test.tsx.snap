// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`brands data should match snapshot 1`] = `
[
  {
    "logo": "VERSACE",
    "name": "Versace",
  },
  {
    "logo": "ZARA",
    "name": "Zara",
  },
  {
    "logo": "GUCCI",
    "name": "Gucci",
  },
  {
    "logo": "PRADA",
    "name": "Prada",
  },
  {
    "logo": "Calvin Klein",
    "name": "Calvin Klein",
  },
]
`;
